### PR TITLE
Implemented clear() for COLLECTIONS-831

### PR DESCRIPTION
**Pull request by Claudenw**
- Defined BloomFilter.clear() to be a reset to unpopulated state.
- Modified BloomFilter implementations.
- Added test case.

Fix for COLLECTIONS-831

### CODE REVIEW

File: src/test/java/org/apache/commons/collections4/bloomfilter/AbstractBloomFilterTest.java [hunk 1]
```diff
@@ -177,6 +177,14 @@ public final void testContains() {
         assertFalse(bf1.contains(bf4));
         assertTrue(bf4.contains(bf1));
     }
+    
+    @Test
+    public void testClear() {
+        BloomFilter bf1 = createFilter(getTestShape(), from1);
+        assertTrue( bf1.cardinality()>0 );
+        bf1.clear();
+        assertTrue( bf1.cardinality()==0);
```
Comment (kinow) [inline src/test/java/org/apache/commons/collections4/bloomfilter/AbstractBloomFilterTest.java:186]: `assertEquals(0, bf1.cardinality())`

**Review by kinow — COMMENTED** [review]
@Claudenw https://issues.apache.org/jira/browse/CASSANDRA-831 ? Is that the correct ticket?

**Review by kinow — APPROVED** [review]

**Comment by kinow** [comment]
Looks good to me. One suggestion for the unit test assertion, but not a blocker IMO. Thanks!

File: src/test/java/org/apache/commons/collections4/bloomfilter/AbstractBloomFilterTest.java [hunk 2]
```diff
@@ -177,6 +177,14 @@ public final void testContains() {
         assertFalse(bf1.contains(bf4));
         assertTrue(bf4.contains(bf1));
     }
+    
+    @Test
+    public void testClear() {
+        BloomFilter bf1 = createFilter(getTestShape(), from1);
+        assertTrue( bf1.cardinality()>0 );
```
Comment (aherbert) [inline src/test/java/org/apache/commons/collections4/bloomfilter/AbstractBloomFilterTest.java:184]: This should use the opposite of assertEquals:
```Java
assertNotEquals(0, bf1.cardinality())
```

**Comment by aherbert** [comment]
The updates have introduced some code duplications. Also you did not update to use `assertNotEquals` in the test for the non-empty filter.

**Comment by aherbert** [comment]
@Claudenw after a lot of commits I think the files are back to the state where we started. So the change to use assertEquals is no longer present, and no change to use assertNotEquals.

**Comment by aherbert** [comment]
Closed by 5a31023eae4bf2bd621fa8613699f54842109402
